src/twisted/internet/endpoints.py

```diff
@@ -808,9 +810,11 @@ def __init__(
             seconds to wait before assuming the connection has failed.
         @type timeout: L{float} or L{int}
 
-        @param bindAddress: the local address of the network interface to make
-            the connections from.
-        @type bindAddress: L{bytes}
+        @param bindAddress: The local address of the network interface to make
+            the connections from, or a (host, port) tuple of local address to
```

Comment:
Can we have the documentation for the types for the `(host, port)` tupple ?

I guess `port` is `int` ... but what about `host` ?

From `def test_tuple(self):` it looks like this is str.

------

Also, what does `local address` mean ?
I think that I know what it means ... but it's not clear if FQDN or hostnames are accepted.

Do we support things other than IPv4 and IPv6 addresses ? 

But maybe this is understood from the context.

Reply:
How does this read?:
```

        @param bindAddress: The client socket normally uses whatever
            local interface (eth0, en0, lo, etc) is best suited for the
            target address, and a randomly-assigned port. This argument
            allows that local address/port to be overridden. Providing
            just an address (as a str) will bind the client socket to
            whichever interface is assigned that address. Providing a
            tuple of (str, int) will bind it to both an interface and a
            specific local port. To bind the port, but leave the
            interface unbound, use a tuple of ("", port), or ("0.0.0.0",
            port) for IPv4, or ("::0", port) for IPv6. To leave both
            interface and port unbound, just use None.
        @type bindAddress: L{str}, L{tuple}, or None
```

Reply:
Thanks. Looks good.